openAgarClient.js

```diff
@@ -5,251 +5,415 @@
+        } else {
+          this.y = Math.max(this.maxY, this.y);
+        }
 
-    function gameLoop() {
-        time = Date.now();
-        frameID = (frameID < 0xFFFFFFFF) ? frameID++ : frameID = 0;
-
-        allNodes.forEach((node) => {
-            node.updatePos();
-            if (node.node) updateNode(node);
-            else drawNode(node);
-        })
-
-        virusNodes.forEach((node) => { // viruses have overlap priority
-            node.updatePos();
-            if (node.node) updateNode(node);
-            else drawNode(node);
-
-        })
-        stage.children.sort(function (a, b) { // sort by size for overlap rules
-            return (a.order || a.size - b.size);
-        })
-        camera()
-            // Draw stuff
-        renderer.render(stage);
-        window.requestAnimationFrame(gameLoop);
+      }
+    }
+    getBounds() {
+      return {
+        x: this.x - this.size,
+        y: this.y - this.size,
+        width: this.size >> 1,
+        height: this.size >> 1
+      }
 
     }
-    setUp()
-    gameLoop();
 
 
+  }
+  allNodes.push(new Node(1, 1, 1, 100, 100))
+  // Main Graphics Setup/loop
+  playerCells.push(allNodes[0])
 
-    function camera() {
-        var total = 0;
-        var tX = 0,
-            tY = 0;
-        playerCells.forEach((node) => {
+  function getScreen() {
+    return {
+      x: window.innerWidth,
+      y: window.innerHeight
+    }
+  }
+
+  function setUp() {
+    /*
+                         +-----------+
+                         |  Camera   |
+                         |           |
+                         +-----+-----+
+                               |
+      +----------------------------------------------+
+      |                        |                     |
+      |                        |                     |
+      |                        |                     |
++-----+-----+            +-----+-----+        +------+------+
+|           |            |           |        |             |
+|   Game    |            |Leaderboard|        |    Chat     |
+|           |            |           |        |             |
++-----------+            +-----------+        +-------------+
+        */
+
+    //Create the renderer
+    //renderer = PIXI.autoDetectRenderer(256, 256);
+
+    let win = getScreen();
+    if (PIXI.utils.isWebGLSupported()) renderer = new PIXI.WebGLRenderer(win.x, win.y, renderOptions);
+    else renderer = new PIXI.CanvasRenderer(win.x, win.y, renderOptions);
+
+    if (!renderer) return alert("Could not establish renderer");
+
+    renderer.backgroundColor = 0xFFFFFF;
+
+    //Add the canvas to the HTML document
+    document.body.appendChild(renderer.view);
+
+    //Create a container object called the `stage`
+    stage = new PIXI.Container();
+
+    // Create camera
+    camera = new PIXI.Container();
+
+	
+    // Create Chat
+    chat.container = new PIXI.Container()
+    let chat_graphics = new PIXI.Graphics();
+    chat_graphics.alpha = .9
```

Comment:
chat_placeholder shouldn't be touched, it's supposed to be an input. I just put text there for an example.